### ISSUE TITLE
High Memory usage in `about:memory`

### ISSUE THREAD

**Issue by Ryman** [bwinton/SnoozeTabs#528]
I'm pretty sure I'm debugging this poorly, but according to `about:memory` this extension's background page seems to have almost 3GB of memory allocated for icons or images? (going by [this page](moz-extension://97faa3dd-ef6d-4c5e-adb4-65f4aae2d154/) referencing this addon) I have around 65 snoozed tabs currently.

![image](https://github.com/bwinton/SnoozeTabs/assets/994978/34e2311d-dd14-43f0-8048-8b335091bb63)

Do you have any insight into how I could help investigate this further, or am I reading this page wrong?

**Comment by bwinton** [comment]
I don't know how you could investigate it further, but I suspect that all the animated gifs probably take up a bunch of memory…Still there should ideally be only 9 of them at any given time, right?

**Comment by jacktose** [comment]
@Ryman I barely understand this, but I'm concerned about reducing Fx memory usage. Is this what you're looking at? about:memory, extension section, filtered for extension guid:
<img width="1662" height="1077" alt="screenshot of about:memory, extension section, filtered for `d17b3bbd-39ba-4427-8a78-51c8c7f1d89c`" src="https://github.com/user-attachments/assets/92469ae0-c086-4451-b173-e152ea196505" />
That's only totaling 12 MB right now. Am I looking at the wrong thing, or just not snoozing tabs with huge images?

**Comment by Ryman** [comment]
> about:memory, extension section, filtered for extension guid

Yep, it should show up there. I imagine it's a mixture of uptime and what kind of favicon's you're encountering. My session's been running for ~3 hours (which aligns with around 180 copies, each from a sync per minute logging if my theory on https://github.com/bwinton/SnoozeTabs/pull/530 is correct):

<img width="1187" height="775" alt="Image" src="https://github.com/user-attachments/assets/c4bd2498-90f3-429a-881b-134d46a52b5e" />

And if I expand the "tiny" section:

<img width="1148" height="1037" alt="Image" src="https://github.com/user-attachments/assets/e51d1cba-e155-46cb-8b5f-6318f719ae0d" />

**Comment by Ryman** [comment]
One thing worth pointing out, when I was debugging this with my build of the extension in a fresh instance, I sometimes saw these allocations in the main process allocation instead of extensions process (but remnants in both). I can only assume that's related to me using the development console in some way?
